python/dgl/heterograph.py

```diff
@@ -3801,81 +3791,45 @@ def adj(self, transpose=False, ctx=F.cpu(), scipy_fmt=None, etype=None):
-
-        Get a scipy coo sparse matrix.
-
-        >>> g.adj(scipy_fmt='coo', etype='develops')
-        <2x3 sparse matrix of type '<class 'numpy.int64'>'
-           with 2 stored elements in COOrdinate format>
-        """
-        etid = self.get_etype_id(etype)
-        if scipy_fmt is None:
-            return self._graph.adjacency_matrix(etid, transpose, ctx)[0]
-        else:
-            return self._graph.adjacency_matrix_scipy(
-                etid, transpose, scipy_fmt, False
-            )
-
-    def adj_sparse(self, fmt, etype=None):
-        """Return the adjacency matrix of edges of the given edge type as tensors of
-        a sparse matrix representation.
-
-        By default, a row of returned adjacency matrix represents the
-        source of an edge and the column represents the destination.
-
-        Parameters
-        ----------
-        fmt : str
-            Either ``coo``, ``csr`` or ``csc``.
-        etype : str or (str, str, str), optional
-            The type names of the edges. The allowed type name formats are:
-
-            * ``(str, str, str)`` for source node type, edge type and destination node type.
-            * or one ``str`` edge type name if the name can uniquely identify a
-              triplet format in the graph.
-
-            Can be omitted if the graph has only one type of edges.
-
-        Returns
-        -------
-        tuple[Tensor]
-            If :attr:`fmt` is ``coo``, returns a pair of source and destination node ID
-            tensors.
-
-            If :attr:`fmt` is ``csr`` or ``csc``, return the CSR or CSC representation
-            of the adjacency matrix as a triplet of tensors
-            ``(indptr, indices, edge_ids)``.  Namely ``edge_ids`` could be an empty
-            tensor with 0 elements, in which case the edge IDs are consecutive
-            integers starting from 0.
-
-        Examples
-        --------
-        >>> g = dgl.graph(([0, 1, 2], [1, 2, 3]))
-        >>> g.adj_sparse('coo')
-        (tensor([0, 1, 2]), tensor([1, 2, 3]))
-        >>> g.adj_sparse('csr')
-        (tensor([0, 1, 2, 3, 3]), tensor([1, 2, 3]), tensor([0, 1, 2]))
+        SparseMatrix(indices=tensor([[0, 1],
+                                     [0, 2]]),
+                     values=tensor([1., 1.]),
+                     shape=(2, 3), nnz=2)
+        >>> g.edata['h'] = {('user', 'follows', 'user'): torch.tensor([3, 2])}
+        >>> g.adj(etype='follows', eweight_name='h')
+        SparseMatrix(indices=tensor([[0, 1],
+                                     [0, 1]]),
+                     values=tensor([3, 2]),
+                     shape=(2, 2), nnz=2)
         """
-        etid = self.get_etype_id(etype)
-        if fmt == "csc":
-            # The first two elements are number of rows and columns
-            return self._graph.adjacency_matrix_tensors(etid, True, "csr")[2:]
+        # Temporal fix to introduce a dependency on torch
+        import torch
```

Comment:
Can this pass UTs of other backends? If not, it's ok to disable them.

Reply:
Skipped other backends.